### PR TITLE
Jd/radial branches

### PR DESCRIPTION
**Pull request by jd-lara**

### CODE REVIEW

File: src/lodf_calculations.jl [hunk 1]
```diff
@@ -268,25 +269,35 @@ function LODF(
     buses::Vector{PSY.ACBus};
     linear_solver::String = "KLU",
     tol::Float64 = eps(),
+    reduce_radial_branches::Bool = false,
```
Comment (daniel-thom) [inline src/lodf_calculations.jl:272]: docstring for this argument?

File: src/radial_braches.jl [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    bus_reduction_map::Dict{Int, Set{Int}}
+    radial_branches::Set{String}
+end
+
+get_bus_reduction_map(rb::RadialBranches) = rb.bus_reduction_map
+get_radial_branches(rb::RadialBranches) = rb.radial_branches
+Base.isempty(rb::RadialBranches) =
+    isempty(rb.bus_reduction_map) && isempty(rb.bus_reduction_map)
+
+function RadialBranches(;
```
Comment (daniel-thom) [inline src/radial_braches.jl:11]: Inner constructor?

File: src/radial_braches.jl [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    line_map::Dict{String, Int},
+    bus_map::Dict{Int, Int},
+    ref_bus_positions::Set{Int})
+    lk = ReentrantLock()
+    buscount = length(bus_map)
+    bus_reduction_map_index = Dict{Int, Set{Int}}()
+    radial_branches = Set{String}()
+    reverse_line_map = Dict(reverse(kv) for kv in line_map)
+    reverse_bus_map = Dict(reverse(kv) for kv in bus_map)
+    #Threads.@threads
```
Comment (daniel-thom) [inline src/radial_braches.jl:114]: Threads are disabled but there is a lock?

File: src/virtual_ptdf_calculations.jl [hunk 4]
```diff
@@ -159,15 +170,16 @@ Checks if the any of the fields of VirtualPTDF is empty.
...
-                   string(getfield(vptdf, name)) * "."
-        elseif !(name in [:K, :dist_slack]) && isempty(getfield(vptdf, name))
-            @debug "Field " * string(name) * " not defined."
-            return true
+        if name == :dist_slack && !isempty(getfield(vptdf, name))
+            @info "Field dist_slack has default value: " *
+                  string(getfield(vptdf, name)) * "."
+            return false
+        elseif (name in [:cache, :radial_branches]) && !isempty(getfield(vptdf, name))
+            @info "Field " * string(name) * " not defined."
```
Comment (daniel-thom) [inline src/virtual_ptdf_calculations.jl:178]: debug? warning?

File: src/virtual_ptdf_calculations.jl [hunk 5]
```diff
@@ -159,15 +170,16 @@ Checks if the any of the fields of VirtualPTDF is empty.
...
 function Base.isempty(vptdf::VirtualPTDF)
     for name in fieldnames(typeof(vptdf))
-        if name == :dist_slack && isempty(getfield(vptdf, name))
-            @debug "Field dist_slack has default value: " *
-                   string(getfield(vptdf, name)) * "."
-        elseif !(name in [:K, :dist_slack]) && isempty(getfield(vptdf, name))
-            @debug "Field " * string(name) * " not defined."
-            return true
+        if name == :dist_slack && !isempty(getfield(vptdf, name))
+            @info "Field dist_slack has default value: " *
```
Comment (daniel-thom) [inline src/virtual_ptdf_calculations.jl:174]: debug?

File: src/radial_braches.jl [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+function RadialBranches(A::IncidenceMatrix)
+    return RadialBranches(A.data, A.lookup[1], A.lookup[2], A.ref_bus_positions)
+end
+
+function RadialBranches(
+    A::SparseArrays.SparseMatrixCSC{Int8, Int64},
+    line_map::Dict{String, Int},
+    bus_map::Dict{Int, Int},
+    ref_bus_positions::Set{Int})
```
Comment (daniel-thom) [inline src/radial_braches.jl:107]: Needs the formatter.

File: src/radial_braches.jl (hunk 2 above)
Comment (jd-lara) [reply src/radial_braches.jl:11]: This is for kwarg build

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**